File: php/tests/ClientTest.php
<?php

namespace AlibabaCloud\SDK\IotApiGateway\Tests;

use AlibabaCloud\SDK\IotApiGateway\Client;
use AlibabaCloud\SDK\IotApiGateway\Models\CommonParams;
use AlibabaCloud\SDK\IotApiGateway\Models\Config;
use AlibabaCloud\SDK\IotApiGateway\Models\IoTApiRequest;
use AlibabaCloud\Tea\Utils\Utils\RuntimeOptions;
use PHPUnit\Framework\TestCase;

/**
 * @internal
 * @coversNothing
 */
class ClientTest extends TestCase
{
    public function testDoRequest()
    {
        $config            = new Config();
        $config->appKey    = getenv('IOT_APP_KEY');
        $config->appSecret = getenv('IOT_APP_SECRET');
        $config->domain    = 'api.link.aliyun.com';

        $client = new Client($config);

        $params = [
            'input' => 'test',
        ];

        $request                = new CommonParams();
        $request->apiVer        = '1.0.0';
        $iotApiRequest          = new IoTApiRequest();
        $iotApiRequest->request = $request;
        $iotApiRequest->params  = $params;

        $runtime = new RuntimeOptions();

        $res    = $client->doRequest(
            '/kit/debug/ping',
            'HTTPS',
            'POST',
            null,
            $iotApiRequest,
            $runtime
        );
        $result = $res->all();
        $this->assertEquals(200, $result['code']);
        $this->assertEquals('test', $result['data']);
    }
}

File: csharp/tests/ClientTest.cs
using System;
using System.Collections.Generic;
using AlibabaCloud.SDK.IotApiGateway;
using AlibabaCloud.SDK.IotApiGateway.Models;
using AlibabaCloud.TeaUtil;
using AlibabaCloud.TeaUtil.Models;
using Tea.Utils;
using Xunit;

namespace tests
{
    public class ClientTest
    {
        [Fact]
        public void TestApi()
        {
            var config = new Config
            { 
                AppKey = Environment.GetEnvironmentVariable("IOT_APP_KEY"),
                AppSecret = Environment.GetEnvironmentVariable("IOT_APP_SECRET"),
                Domain = "api.link.aliyun.com"
            };

            var client = new Client(config);

            var param = new Dictionary<string, string>
            {
                { "input", "test" }
            };

            var req = new CommonParams
            {
                ApiVer = "1.0.0"
            };

            var body = new IoTApiRequest
            {
                Params = param,
                Request = req
            };

            var runtime = new RuntimeOptions();

            var resp = client.DoRequest("/kit/debug/ping", "HTTPS", "POST", null, body, runtime);

            var obj = AlibabaCloud.TeaUtil.Common.ReadAsJSON(resp.Body);

            var map = AlibabaCloud.TeaUtil.Common.AssertAsMap(obj);

            Assert.Contains("test", map.Get("data").ToSafeString());
        }
    }
}
